### PR TITLE
Adjust the trace structure.

### PR DESCRIPTION
**Pull request by jmjoy**
To add logging and metrics in the future, adjust the structure of the trace.

1. Rename mod `context` to `trace`.
2. <del>Move `reporter` to `trace`, and rename `Reporter` to `TraceReporter`, and do so for implementers.</del>
3. Add the `logging` mod and base structure.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -19,7 +19,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         .build_server(false)
         .type_attribute(".", "#[derive(serde::Serialize, serde::Deserialize)]")
         .compile(
-            &["./skywalking-data-collect-protocol/language-agent/Tracing.proto"],
+            &[
+                "./skywalking-data-collect-protocol/language-agent/Tracing.proto",
+                "./skywalking-data-collect-protocol/logging/Logging.proto",
```
Comment (wu-sheng) [inline build.rs:24]: This is for further using, right?

Comment (wu-sheng) [reply build.rs:24]: I think log report doesn't implement in the PR.

**Comment by wu-sheng** [comment]
> Move reporter to trace, and rename Reporter to TraceReporter, and do so for implementers.

Mostly, we recommend only having one global reporter for trace, log, and metrics.

File: build.rs (hunk 1 above)
Comment (jmjoy) [reply build.rs:24]: Yes, is better to restore it and add it in future?

**Comment by jmjoy** [comment]
> > Move reporter to trace, and rename Reporter to TraceReporter, and do so for implementers.
> 
> Mostly, we recommend only having one global reporter for trace, log, and metrics.

Alright.

File: build.rs (hunk 1 above)
Comment (wu-sheng) [reply build.rs:24]: It is fine, I just want to confirm this. It is OK to keep.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**